### ISSUE TITLE
Error during add integration

### ISSUE THREAD

**Issue by fsda9876** [dboeni/home-assistant-ampere-storage-pro-modbus#6]
Hello,
first thank you for your exciting work. I was searching very intense how the Amper.Storage can be integrated in some HomeAutomation system. Trying to test the integration I ended up in the following error:

`Config flow could not be loaded: {"message":"Invalid handler specified"}`

![image](https://github.com/user-attachments/assets/974f146d-2075-4db9-9e8c-3d4e4baff065)

I run HASS in WSL (installed after this: https://www.home-assistant.io/installation/windows#install-wsl, with version 2024.10.3)

I installed the component as descripted via HACS and verified that all files in custom_components folder:

![image](https://github.com/user-attachments/assets/44ba77e2-3aee-464c-91a3-7ed110709505)

In the logs I found:
`2024-10-20 23:43:19.950 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration ampere_modbus: No module named 'pymodbus.register_read_message'`

Checking Python-Environment pymodbus is installed:
```
(homeassistant) homeassistant@Computer:~$ pip list pymodbus
Package Version
------- -------
pip     24.0
```

Are there some requirements missing? I'm new to HASS - are missing requirements added automatically if a new custom component is added?

Thank you for your help in advance!

**Comment by fsda9876** [comment]
I found the issue - in hub.py the import has to be changed to:
`from pymodbus.pdu.register_read_message import ReadInputRegistersResponse`
After that I could import it. 

But nevertheless - is there something to enable. Connection from inverter is refused.
```
2024-10-21 14:58:43.722 ERROR (SyncWorker_3) [custom_components.ampere_modbus.hub] Reading inverter data
2024-10-21 14:58:43.729 ERROR (SyncWorker_3) [pymodbus.logging] Connection to (192.168.2.87, 502) failed: [Errno 111] Connection refused
2024-10-21 14:58:43.730 ERROR (MainThread) [custom_components.ampere_modbus.hub] Reading realtime data failed! Inverter is unreachable.
```

**Comment by fsda9876** [comment]
Some additional comment - I tested it on an Raspi, too. It seems there are different version used of pymodbus. Here the initial import is working.

**Comment by dboeni** [comment]
Thanks for the hint.
Did you install the component manually (copy & paste) or by adding the repository in HACS?

**Comment by dboeni** [comment]
Information about the REST API:
The Ampere.IQ-BOX provides an API. There is a yaml for HA for this:
https://github.com/dboeni/home-assistant-ampere-storage-pro

However, I will only focus on and develop the Modbus variant

**Comment by fsda9876** [comment]
For WSL I installed it via HACS and cross-checked it where it is installed. I just checked with pip the packages installed (I didn’t install additional packages – so I think HACS take care – I did see the requirements in the manifest). The modification I made in place.

On Raspi, I used the Backup from WSL and modified in place.

 

 

Thanks for the hint.
Did you install the component manually (copy & paste) or by adding the repository in HACS?
